packages/framework/lib/handlers/text.ts

```diff
@@ -35,12 +35,15 @@ export default function textHandlerFactory<U extends User<any>>(
 ) {
     function textHandler(
         this: Bot<U>,
-        message: { text: string },
+        message: { text: string; data?: { [key: string]: any } },
```

Comment:
why do we need that extra data field? what are we inputting to this? Couldn't you just pass the nlp object to the action?

Reply:
The data field is needed for the tracking data enhancement.
We have the ability to pass a data object: { [key: string]: any } regardless if it is a post back tracking_data or not.
This way each function that response to a text now has the information of the text written by the user (or send by a button press)  as well as the tracking data we provided on previews calls.